pkg/hubble/exporter/config_watcher.go

```diff
@@ -57,69 +66,29 @@ func (c *configWatcher) watch(ctx context.Context, interval time.Duration) error
 }
 
 func (c *configWatcher) reload() {
-	c.logger.Debug("Attempting reload")
-	config, hash, err := c.readConfig()
+	configs, hash, err := c.parseConfig()
 	if err != nil {
 		DynamicExporterReconfigurations.WithLabelValues("failure").Inc()
-		c.logger.WithError(err).Warn("failed reading dynamic exporter config")
-	} else {
-		c.callback(hash, *config)
+		c.logger.WithError(err).Error("Failed to parse dynamic exporter config")
+		return
 	}
+	c.callback(configs, hash)
 }
 
-func (c *configWatcher) readConfig() (*DynamicExportersConfig, uint64, error) {
-	config := &DynamicExportersConfig{}
-	yamlFile, err := os.ReadFile(c.configFilePath)
+func (c *configWatcher) parseConfig() (map[string]ExporterConfig, uint64, error) {
+	content, err := os.ReadFile(c.configFilePath)
```

Comment:
You can  use `os.Open` and pass it directly to c.configParser.Parse instead of reading into a buffer.

Reply:
I need to comoute the hash with the content

Reply:
Ah I see. 👍